web/src/stream_events.ts

```diff
@@ -146,6 +146,9 @@ export function update_property<P extends keyof UpdatableStreamProperties>(
             stream_muting.update_is_muted(sub, value, message_view.rerender_combined_feed);
             stream_list.refresh_muted_or_unmuted_stream(sub);
             recent_view_ui.complete_rerender();
+            // Since muted streams aren't counted in visible unread
+            // counts, we need to update the rendering of them.
+            unread_ui.update_unread_counts();
```

Comment:
I think it might make more sense to either have all of this code be in `stream_muting.update_is_muted` or no UI update code in that other module, i.e. inlining it here and deleting that one-function file.

The main thing that's messy is that `message_lists.all_rendered_message_lists` loop, and that surely could be made a function in `message_live_update.ts`.

Would be a good little prep PR.

Reply:
If that worked, I can also make that a prep PR

Reply:
Yup it worked! #35107

Reply:
Okay for the sake of time I'm going to remove this commit and leave it as it is, and add `unread_ui.update_unread_counts();` to `build_stream_list` for this PR since that ensures the stream list gets built correctly. I can look into this wider refactor in the future.